module/finalizer/collection/finalizer_test.go

```diff
@@ -104,8 +103,8 @@ func TestFinalizer(t *testing.T) {
 			bootstrap()
 			defer cleanup()
 
-			prov := new(mocknetwork.Engine)
-			prov.On("SubmitLocal", mock.Anything)
+			prov := new(collectionmock.GuaranteedCollectionPublisher)
```

Comment:
```suggestion
			prov := collectionmock.NewGuaranteedCollectionPublisher(t)
```
The testing library we use for mocks provides some extra functionality when we use the constructor instead of `new`. It will register assertion checks at the end of the test, so if we call `prov.On("SubmitCollectionGuarantee", mock.Anything)` during the test, the library will ensure that asserted function was called, otherwise it will immediately fail the test. Do you mind updating instances in this test to use the constructor form?

(It doesn't make much difference in this test, but will help for more complex test cases in the future.)